src/app/login/page.tsx

```diff
@@ -4,27 +4,69 @@ import { useCallback, useRef } from "react";
 
 import { Button } from "@/components/ui/button";
 import { Input } from "@/components/ui/input";
+
 import { useNostrContext } from "@/lib/nostr/NostrContext";
+import { LoginType, checkType } from "@/lib/utils";
 
 import { Puzzle } from "lucide-react";
 import Image from "next/image";
 import { useRouter } from "next/navigation";
 
+import { nip19, nip05 } from "nostr-tools"
+
+
 export default function Login() {
+
   const { setAuthor } = useNostrContext();
   const router = useRouter();
+
   const inputRef = useRef<HTMLInputElement>(null);
 
-  /* try entering npub18zx8lw3947pghsgzqv2t0x8pe767sscag5djgj5afr755xkqd97qt530pr */
+  // TODO : setAuthor needs to be tweaked (don't remove but tweak *_*)
 
-  const handleSubmit = useCallback(
-    (e: React.FormEvent<HTMLFormElement>) => {
-      e.preventDefault();
-      setAuthor && setAuthor(inputRef.current?.value || "");
-      router.push("/");
-    },
-    [setAuthor, router]
-  );
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+
+    e.preventDefault();
+    await handleLogin();
+    console.log("HandleSubmit");
+
+  };
+
+  const handleLogin = useCallback(async () => {
+
+    const cred = inputRef.current?.value || '';
+    let loginType = checkType(cred);
+    let npub = '';
+
+    console.log("Yay!")
+
+    // Checking for nip07 extension
+    loginType = (window.nostr && cred === '') ? LoginType.nip07 : loginType; 
+    
+    // Else use common npub credentials
```

Comment:
NIT: I wonder if we wanna react-hook-form https://react-hook-form.com/

Reply:
I'm not a front-end developer, can't help 🤣🤣🤣🤣🤣, based on the website, it might be a good idea, but I'm concerned about bloat, is it supported on vanilla react ?